### ISSUE TITLE
select: Naming of <selectedoption>

### ISSUE THREAD

**Issue by jakearchibald** [openui/open-ui#1112]
Sorry, I know bikeshedding is painful, but it's really awkward that "the `<selectedoption>`" and "the selected `<option>`" are different things, even though they sound the same verbally.

Try saying "the `<selectedoption>` contains a clone of the content of the selected `<option>`" out loud. You need to say something like "the `<selectedoption>` contains a clone of the content of the `<option>` that is selected" to make it understandable, but that also sounds weird.

It's a problem in code too:

```js
const selectedOption = select.selectedOptions[0];
const selectedOption = select.querySelector('selectedoption');
```

**Comment by lukewarlow** [comment]
For context here's the original discussion on naming: https://github.com/openui/open-ui/issues/808

There were a few proposals. I liked **value** or **selectedvalue** but it was pointed out that's not quite right because it's not actually the value. **selected** is perhaps too ambiguous?

There's no prior art from which to base the naming off unfortunately.

Potentially worth considering what might the element look like for select multiple in future?

**Comment by jakearchibald** [comment]
`<selectedcontent>` is the best I've got.

**Comment by brucelawson** [comment]
<selection> ?

On Fri, 18 Oct 2024 at 12:41, Jake Archibald ***@***.***> wrote:
>
> <selectedcontent> is the best I've got.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



-- 
Bruce Lawson
www.brucelawson.co.uk
@***@***.***

**Comment by smhigley** [comment]
`<selection>` feels a little too close to text selection to me 😅

I kinda like `<selectedcontent>` in that it makes it a little more intuitive that it's a semantically generic container mirroring the content of the selected option, and not literally a semantic option nested inside the button.

**Comment by jakearchibald** [comment]
I've seen a few folks that think `<selectedoption>` is a replacement for `<option selected>`, which is totally understandable.

**Comment by josepharhar** [comment]
I agree, `<selectedoption>` verbally conflates with "selected `<option>`" really badly. `<selectedcontent>` sounds good to me.

**Comment by sorvell** [comment]
What about repurposing the [`<output>` element](https://html.spec.whatwg.org/multipage/form-elements.html#the-output-element) for this?

It probably doesn't work, but it seems at least conceptually similar since it's primarily displaying the result of a user action.

**Comment by jakearchibald** [comment]
That feels like a pretty big semantic switch.

**Comment by sorvell** [comment]
[claude.ai](//claude.ai) thinks `choice` is the strongest option:

```html
<select>
  <button>
    <choice>Apple</choice> <!-- automatically updates when selection changes -->
  </button>
  <option value="apple">Apple</option>
  <option value="banana">Banana</option>
</select>
```

`choice` works well because it:
1. Is distinct from other select-related terms
2. Clearly represents its purpose
3. Is short and simple
4. Isn't easily confused with existing properties
5. Makes semantic sense in the context

**Comment by josepharhar** [comment]
Here is our previous thread on naming, where we renamed it from `<selectedvalue>` to `<selectedoption>`: https://github.com/openui/open-ui/issues/808

I'd rather not use `<output>` for this since it already has semantics/behaviors.

I like `<selectedcontent>` because it represents the content of the currently selected option. `<choice>` sounds a bit vague by comparison.

**Comment by brechtDR** [comment]
+1 on slectedcontent
